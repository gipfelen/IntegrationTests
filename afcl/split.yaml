---
name: "split"
dataIns:
  - name: "int_array"
    type: "collection"
    source: "int_array"
  - name: "chunks"
    type: "number"
    source: "chunks"
workflowBody:
  - parallelFor:
      name: "forEach"
      dataIns:
        - name: "in"
          type: "collection"
          source: "split/int_array"
          constraints:
            - name: "split"
              value: "split/chunks"
      iterators: ["in"]
      loopBody:
        - function:
            name: "function"
            type: "SumCollection"
            dataIns:
              - name: "collectionToSum"
                type: "collection"
                source: "forEach/in"
              - name: "waitTimeIn"
                type: "number"
                source: 1000
            dataOuts:
              - name: "collectionSum"
                type: "number"
      dataOuts:
        - name: "result"
          type: "collection"
          source: "function/collectionSum"
dataOuts:
  - name: "result"
    type: "collection"
    source: "forEach/result"
