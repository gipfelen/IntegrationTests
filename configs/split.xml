<configuration>
  <module class="at.uibk.dps.ee.control.modules.EnactmentVerticleModule"/>
  <module class="at.uibk.dps.ee.io.modules.InputReaderFileModule">
    <property name="filePath">./inputData/split.json</property>
  </module>
  <module class="at.uibk.dps.ee.io.modules.OutputPrinterModule"/>
  <module class="at.uibk.dps.ee.io.modules.SpecificationInputModule">
    <property name="filePathAfcl">./afcl/split.yaml</property>
    <property name="filePathMappingFile">./typeMappings/split.json</property>
  </module>
  <module class="at.uibk.dps.ee.visualization.modules.EnactmentViewerModule">
    <property name="closeOnTerminate">false</property>
  </module>
  <module class="at.uibk.dps.sc.core.modules.SchedulerModule"/>
</configuration>
