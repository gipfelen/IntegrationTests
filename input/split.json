{
	"int_array" : [1,2,3,4],
	"chunks" : 2
}