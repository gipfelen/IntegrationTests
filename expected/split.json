{"result":[3,7]}